server/blob/blob-union/src/main/java/org/apache/james/blob/union/HybridBlobStore.java

```diff
@@ -128,43 +139,59 @@ private HybridBlobStore(BlobStore lowCostBlobStore, BlobStore highPerformanceBlo
 
     @Override
     public Mono<BlobId> save(BucketName bucketName, byte[] data, StoragePolicy storagePolicy) {
-        return selectBlobStore(storagePolicy, Mono.just(data.length > configuration.getSizeThreshold()))
-            .flatMap(blobStore -> blobStore.save(bucketName, data, storagePolicy));
+        return Flux.from(selectBlobStores(storagePolicy, Mono.just(data.length > configuration.getSizeThreshold())))
+            .flatMap(blobStore -> blobStore.save(bucketName, data, storagePolicy))
+            .distinct()
+            .single();
```

Comment:
Don't single will throw an `IndexOutOfBoundsException` if there are two blobstores because of duplicateWrite ?

Reply:
Both blobStores are expected to store the same blob hance distinct should keep a single blobId, however I would like us to explicitly throw a human understandable exception in case the content stored on both blob stores differ.